packages/manager/apps/web/client/app/components/alldom/domain-alldom.service.js

```diff
@@ -97,4 +99,23 @@ export default class {
       data,
     });
   }
+
+  /**
+   * Get domains of alldom pack and its serviceInfo
+   * @param {string} serviceName
+   */
+  async getDomainsWithServiceInfo(serviceName) {
+    const domains = await this.getDomains(serviceName);
+    const promises = domains.map((domain) =>
+      this.$http.get(`/domain/${domain}/serviceInfos`),
+    );
+    return this.$q.all(promises).then((serviceInfos) => {
+      return domains.map((domain, index) => {
+        return {
+          name: domain,
+          serviceInfo: serviceInfos[index].data,
+        };
+      });
+    });
```

Comment:
to be consistent with the rest we can use chaining promises :

```suggestion
    return this.getDomains(serviceName)
    .then((domains) => domains.map((domain) => this.$http.get(`/domain/${domain}/serviceInfos`)))
    .then((domains) => this.$q.all(promises).then((serviceInfos) => {
      return domains.map((domain, index) => {
        return {
          name: domain,
          serviceInfo: serviceInfos[index].data,
        };
      });
    }))
```